src/Workspaces/Core/Portable/LinkedFileDiffMerging/LinkedFileDiffMergingAggregatedLogger.cs

```diff
@@ -0,0 +1,200 @@
+
+        private object instanceLock = new object();
+        private readonly List<LinkedFileGroupSessionInfo> groupsWithMergeComments = new List<LinkedFileGroupSessionInfo>();
+        internal int totalSessions;
+        internal int totalSessionsWithLinkedFiles;
+        internal int totalLinkedFileGroupsProcessed;
+        internal int totalIdenticalDiffs;
+        internal int totalIsolatedDiffs;
+        internal int totalOverlappingDistinctDiffs;
+        internal int totalOverlappingDistinctDiffsWithSameSpan;
+        internal int totalOverlappingDistinctDiffsWithSameSpanAndSubstringRelation;
+        internal int totalInsertedMergeConflictComments;
+        internal int totalInsertedMergeConflictCommentsAtAdjustedLocation;
+
+
+        internal static void LogSession(Workspace workspace, LinkedFileDiffMergingSessionInfo sessionInfo)
+        {
+            lock (globalLock)
+            {
+                if (!loggers.ContainsKey(workspace))
+                {
+                    loggers[workspace] = new LinkedFileDiffMergingAggregatedLogger();
+                    var coordinator = workspace.Services.GetService<ITelemetryAggregationCoordinator>();
+                    coordinator.RegisterSolutionClosedLogger(loggers[workspace]);
+                }
+
+                var logger = loggers[workspace];
+                logger.LogSession(sessionInfo);
+            }
+        }
+
+        internal void LogSession(LinkedFileDiffMergingSessionInfo sessionInfo)
+        {
+            lock (instanceLock)
+            {
+                totalSessions++;
+
+                if (sessionInfo.LinkedFileGroups.Any())
+                {
+                    totalSessionsWithLinkedFiles++;
+                    totalLinkedFileGroupsProcessed += sessionInfo.LinkedFileGroups.Count;
+
+                    foreach (var groupInfo in sessionInfo.LinkedFileGroups)
+                    {
+                        totalIdenticalDiffs += groupInfo.IdenticalDiffs;
+                        totalIsolatedDiffs += groupInfo.IsolatedDiffs;
+                        totalOverlappingDistinctDiffs += groupInfo.OverlappingDistinctDiffs;
+                        totalOverlappingDistinctDiffsWithSameSpan += groupInfo.OverlappingDistinctDiffsWithSameSpan;
+                        totalOverlappingDistinctDiffsWithSameSpanAndSubstringRelation += groupInfo.OverlappingDistinctDiffsWithSameSpanAndSubstringRelation;
+                        totalInsertedMergeConflictComments += groupInfo.InsertedMergeConflictComments;
+                        totalInsertedMergeConflictCommentsAtAdjustedLocation += groupInfo.InsertedMergeConflictCommentsAtAdjustedLocation;
+
+                        if (groupInfo.InsertedMergeConflictComments > 0 ||
+                            groupInfo.InsertedMergeConflictCommentsAtAdjustedLocation > 0)
+                        {
+                            groupsWithMergeComments.Add(groupInfo);
```

Comment:
so, we hold groupsWithMergeComments for entire life time of Solution? why not log it right away?